common/djangoapps/util/db.py

```diff
@@ -142,6 +146,29 @@ def commit_on_success(using=None, read_committed=False):
         return CommitOnSuccessManager(using, read_committed)
 
 
+@contextmanager
+def enable_outer_atomic(names):
+    """
+    Enable outer_atomics with names.
+
+    Can be used either as a decorator or a context manager.
+    See docstring of outer_atomic for details.
+
+    Arguments:
+        names (a string or list of strings): Names of outer_atomics.
+    """
+    if isinstance(names, basestring):
```

Comment:
Ah, yes, that would be a bit nicer. Though multiple names are going to be extremely rare.